src/OrleansCodeGenerator/CodeGeneratorCommon.cs

```diff
@@ -87,9 +89,13 @@ public static byte[] CompileAssembly(GeneratedSyntax generatedSyntax, string ass
                     .AddSyntaxTrees(code.SyntaxTree)
                     .AddReferences(assemblies)
                     .WithOptions(options);
+            MemoryStream symbolStream = null;
             using (var stream = new MemoryStream())
+#if DEBUG
+            using (symbolStream = new MemoryStream())
+#endif
```

Comment:
Should we try to dynamically determine if it should use debug symbols if the *target* assembly is debuggable instead? Otherwise it will be almost always the case that people use the official nuget packages that were compiled in release mode.
There is a method that we are currently using called `RuntimeVersion.IsAssemblyDebugBuild` that *may* be useful for this test (after making it internal).